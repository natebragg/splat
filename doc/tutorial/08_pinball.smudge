// 08_pinball.smudge
pinball
{
   * initialize -(@lockPaddles)-> idle,
     idle
     [
          coin --> prepare-ball
     ],
     prepare-ball (@releaseBall, @resetScore, @unlockPaddles)
     [
          plunger --> run
     ],
     run (@startSound)
     [
          target -(@incScore)-,
          drain -(@sadSound, @displayScore)-> idle
     ] (@lockPaddles),
     lockout (@resetScore, @displayError, @startTimer)
     [
          timerExpired --> idle,
          tilt --,
          _ --
     ] (@cancelTimer),
     _
     [
          tilt --> lockout,
          coin -(@rejectCoin)-
     ]
}
