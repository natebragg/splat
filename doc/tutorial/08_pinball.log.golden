Locking paddles.
Dropping ball.
Unlocking paddles. Whirrrrrr
Ding ding ding ding ding
Coin at a bad time. Dropping it.
DingDingDingDing
Awwwwwww
Score: 4
A new high score!
Locking paddles.
Dropping ball.
Unlocking paddles. Whirrrrrr
Ding ding ding ding ding
DingDingLocking paddles.
TILT!
Coin at a bad time. Dropping it.
